### DISCUSSION TITLE
Performance Issues with Large Library

### DISCUSSION THREAD

**Discussion by Brajesh3** [komikku-app/komikku#497] [Q&A] [answered]
# **Problem**
I am using komikku Stable with a library containing over 450 items. When I go to the **Browse** section and select any extension, it takes over 30 seconds to load, and scrolling also causes repeated delays.

# **Testing**
I tested this with Comick Preview without restoring my library (0 items). In this case, browsing opens instantly, and scrolling loads content very quickly (under 5 seconds).
> [!Note]
> When I restore my library in Comick Preview (Except app settings), the same behavior occurs with default setting.

# **Questions**
1. Is this behavior expected with a large library?
2. Is the slowness caused by the size of my library?
3. Could there be a misconfiguration on my end?
4. Are there any settings I could adjust to improve performance?

**Comment by cuong-tran** [comment]
My library is 21000 entries.

**Reply by Brajesh3** [reply]
Are you talking about chapters
![Screenshot_20241103-175317_Komikku](https://github.com/user-attachments/assets/8f8946a0-83e7-487b-b10b-461ef95cfbcc)

**Reply by cuong-tran** [reply]
![Screenshot_20241103_194830_Komikku.jpg](https://github.com/user-attachments/assets/3e446ec2-8bca-4e58-a318-2acdc120cc56)

**Reply by Turhvjbufv** [reply]
Those are rookie numbers you gotta pump these up

**Comment by cuong-tran** [comment]
Anyway, it does require access to database while browsing. Slowly accessing db could because of phone storage speed. Are you using external SD card or do you have a full storage?

**Reply by Brajesh3** [reply]
app installed internally
storage location is sd card

**Reply by cuong-tran** [reply, marked as answer]
Try internal storage

**Reply by Brajesh3** [reply]
thanks now it's working as intended
